pkgs/development/libraries/science/networking/ns3/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, fetchFromGitHub, fetchurl
+, python
+
+# for binding generation (untested)
+, castxml ? null, pygccxml ? null
+
+# for documentation
+, doxygen ? null, graphviz ? null, imagemagick ? null
+
+# for manual, tetex is used to get the eps2pdf binary
+# texlive to get latexmk. building manual still fails though
+, dia, tetex ? null, ghostscript ? null, texlive ? null
+
+, withDoc ? false
+, withManual ? false
+, withGsl ? false
+
+# --enable-examples
+, withExamples ? false
+
+# --enable-tests
+, withTests ? true
+, generateBindings ? false
+
+# All modules can be enabled by choosing 'all_modules'.
+# included here the DCE mandatory ones
+, modules ? [ "core" "network" "internet" "point-to-point"]
+, pkgs
```

Comment:
Please do not add `pkgs` and pass `gcc6`, `python` and `castxml` directly here.

Reply:
you already warned me on that for other PRs. Sorry. I used to start writing the packages  from the overlay, hence the pkgs remaining from "super.pkgs" which I forgot to remove. It's more convenient then writing `callPackage ../ns3 { inherit (super.pkgs) gcc6 python castxml }`. In fact I am curious how people create their package while waiting for a merge. If I add it to nixpkgs, then I lose the package once switching  to another branch so it's hard to develop several packages at the same time.

Reply:
I personally use a nixpkgs checkout instead of channels: http://anderspapitto.com/posts/2015-11-01-nixos-with-local-nixpkgs-checkout.html